### PR TITLE
Dev/alha

### PR DESCRIPTION
**Pull request by alexjhawk**

### CODE REVIEW

**Comment by alexjhawk** [comment]
Due to limitations with the message error handling, I could not easily detect that OSIsoft was returning an error. Without this context, messages appeared entirely successful until data was noticed missing on the server.

I have significantly upgraded the response handling of this connector so incidents such as this can be caught and remedied much sooner.

**Review by oliver-walker-hms — APPROVED** [review]

File: src/main/java/com/hms/flexyosisoftconnector/dataserver/OSIsoftServer.java [hunk 1]
```diff
@@ -108,72 +108,35 @@ public static boolean RequestHTTPS(
...
-        if (jsonStartIndex == notFoundIndex) {
-          connected = true;
-          Logger.LOG_SERIOUS("Response received from OSIsoft server was not valid!");
-          return connected;
-        }
+    // Remove random characters before start of JSON in response file
+    String jsonStart = "{";
+    if (responseFileString != null
+        && !responseFileString.startsWith(jsonStart)
+        && responseFileString.indexOf(jsonStart) != -1) {
```
Comment (TomKimsey) [inline src/main/java/com/hms/flexyosisoftconnector/dataserver/OSIsoftServer.java:124]: Magic number

File: src/main/java/com/hms/flexyosisoftconnector/dataserver/OSIsoftServerResponseUtil.java [hunk 2]
```diff
@@ -0,0 +1,427 @@
...
+    }
+  }
+
+  /**
+   * Handles a messages array in an HTTPS response from the OSIsoft server.
+   *
+   * @param response The response from the OSIsoft server to handle
+   * @param connectionUrl The URL of the OSIsoft server request that was made
+   * @param responseFileName The name of the file which the response was saved to
+   * @return boolean indicating if an error severity message was found. If no error severity message
```
Comment (TomKimsey) [inline src/main/java/com/hms/flexyosisoftconnector/dataserver/OSIsoftServerResponseUtil.java:228]: The response value and variable name seem strange. 

"@return boolean indicating if an error severity message was found." leads you to initially thing true = error.

also why store an inverted boolean? (noErrorSeverityFound) when its much more typical to store a positive value [isErrorSeverityFound].

This inversion just seems strange and prone to user error.

**Review by TomKimsey — CHANGES_REQUESTED** [review]

File: src/main/java/com/hms/flexyosisoftconnector/dataserver/OSIsoftServer.java (hunk 1 above)
Comment (alexjhawk) [reply src/main/java/com/hms/flexyosisoftconnector/dataserver/OSIsoftServer.java]: Fixed

File: src/main/java/com/hms/flexyosisoftconnector/dataserver/OSIsoftServerResponseUtil.java (hunk 2 above)
Comment (alexjhawk) [reply src/main/java/com/hms/flexyosisoftconnector/dataserver/OSIsoftServerResponseUtil.java]: Updated Javadoc and inverted the boolean to `foundMessageWithErrorSeverity`

**Review by TomKimsey — APPROVED** [review]

**Comment by alexjhawk** [comment]
@oliver-walker-hms Can you re-approve with latest changes?

**Review by oliver-walker-hms — APPROVED** [review]

**Pull request merged**